env0/data_google_cost_credentials.go

```diff
@@ -0,0 +1,93 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataGoogleCostCredentials() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataGoogleCostCredentialsRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Description:  "the name of the credential",
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+			},
+			"id": {
+				Type:         schema.TypeString,
+				Description:  "the id of the credential",
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+			},
+		},
+	}
+}
+
+func dataGoogleCostCredentialsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var err diag.Diagnostics
+	var credentials client.Credentials
+
+	id, ok := d.GetOk("id")
+	if ok {
+		credentials, err = getGoogleCostCredentialsById(id.(string), meta)
+		if err != nil {
+			return err
+		}
+	} else {
+		name, _ := d.Get("name").(string) // name must be specified here
+		credentials, err = getGoogleCostCredentialsByName(name, meta)
+		if err != nil {
+			return err
+		}
+	}
+
+	d.SetId(credentials.Id)
+	d.Set("name", credentials.Name)
+
+	return nil
+}
+
+func getGoogleCostCredentialsByName(name interface{}, meta interface{}) (client.Credentials, diag.Diagnostics) {
+	apiClient := meta.(client.ApiClientInterface)
+	credentialsList, err := apiClient.CloudCredentialsList()
+	if err != nil {
+		return client.Credentials{}, diag.Errorf("Could not query GCP Credentials by name: %v", err)
+	}
+
+	credentialsByNameAndType := make([]client.Credentials, 0)
+	for _, candidate := range credentialsList {
+		if candidate.Name == name.(string) && isValidGoogleCostCredentialsType(candidate.Type) {
+			credentialsByNameAndType = append(credentialsByNameAndType, candidate)
+		}
+	}
+
+	if len(credentialsByNameAndType) > 1 {
+		return client.Credentials{}, diag.Errorf("Found multiple Google cost Credentials for name: %s", name)
```

Comment:
Please align this error message with all other error messages in this file...

Reply:
4eec18f9488a26f0acf5fe31b46a44b23cc303cd